### PR TITLE
handshake: rewrite sequences for clarity and correctness

### PR DESCRIPTION
**Pull request by ttaylorr**

### CODE REVIEW

File: handshake/client_ack_sequence.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+	return nil
+}
+
+// Write implements the Sequence.Write function. It writes the S1 packet first
+// (returning any errors if there is one), and then writes the S2 packet with
+// the same data as was sent in the C1 packet (returning any error that was
+// encountered).
+//
+// A successful call to Write constitutes a value of `nil` being returned.
+func (c *ClientAckSequence) Write(w io.Writer) error {
```
Comment (connor4312) [inline handshake/client_ack_sequence.go]: Semantically you normally want .WriteTo

Comment (ttaylorr) [inline handshake/client_ack_sequence.go]: Fair, I'll change that.

**Comment by connor4312** [comment]
looks good overall

**Comment by ttaylorr** [comment]
Thanks for the review! 👋

**Pull request merged**